File: src/main/java/gameplay/Oval.java
package gameplay;

public class Oval implements Shape {
    public double x, y;
    private double radiusX, radiusY;

    public Oval(Oval o) {
        this.x = o.x;
        this.y = o.y;
        this.radiusX = o.radiusX;
        this.radiusY = o.radiusY;
    }

    public Oval(Point pos, int radius) {
        this(pos, radius, radius);
    }

    public Oval(Point pos, int radiusX, int radiusY) {
        this.x = pos.x;
        this.y = pos.y;
        this.radiusX = radiusX;
        this.radiusY = radiusY;
    }

    public int getX() {
        return (int) x;
    }

    public int getY() {
        return (int) y;
    }

    public int getRadiusX() {
        return (int) radiusX;
    }

    public int getRadiusY() {
        return (int) radiusY;
    }

    @Override
    public boolean contains(Point p) {
        return Math.pow(p.x - this.x, 2) / Math.pow(this.radiusX, 2) +
                Math.pow(p.y - this.y, 2) / Math.pow(this.radiusY, 2) <= 1;
    }
}

File: src/main/kotlin/gameplay/geometry/Oval.kt
package gameplay.geometry

import gameplay.Point
import kotlin.math.pow

class Oval(private val x: Double,private val y: Double, private val radiusX: Double, private val radiusY: Double) : Shape {

    constructor(o: Oval) : this(o.x, o.y, o.radiusX, o.radiusY)
    constructor(pos: Point, radiusX: Int, radiusY: Int) : this(pos.x, pos.y, radiusX.toDouble(), radiusY.toDouble())

    val center: Point
        get() = Point(x, y)

    fun getX(): Int {
        return x.toInt()
    }

    fun getY(): Int {
        return y.toInt()
    }

    fun getRadiusX(): Int {
        return radiusX.toInt()
    }

    fun getRadiusY(): Int {
        return radiusY.toInt()
    }

    override operator fun contains(point: Point): Boolean {
        return (point.x - x).pow(2.0) / radiusX.pow(2.0) +
                (point.y - y).pow(2.0) / radiusY.pow(2.0) <= 1
    }
}
